lib/renderer/html/index.js

```diff
@@ -366,10 +338,24 @@ HTMLRenderer.prototype.createElement = function(node, path, entityId){
   }
 
   if (node.type === 'component') {
-    var fragment = document.createDocumentFragment();
     var child = new Entity(node.component, node.props);
-    var el = this.mountEntity(child, fragment);
     this.children[entityId][path] = child.id;
-    return el;
+    return this.mount(child);
+  }
+};
+
+/**
+ * Render the entity and make sure it returns a node
+ *
+ * @param {Entity} entity
+ *
+ * @return {VirtualTree}
+ */
+
+function renderEntity(entity) {
+  var result = entity.render();
+  if (!result) {
+    throw new Error('Render function must return a virtual node.');
```

Comment:
Maybe `virtualize node` would be more correct, since that's the library being used.